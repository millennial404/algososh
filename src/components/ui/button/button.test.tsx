import React from 'react';
import {render} from '@testing-library/react';
import {Button} from './button';

describe('Button component', () => {
  test('корректно отображает кнопку с текстом', () => {
    const {container} = render(<Button text="Click me!"/>);
    expect(container).toMatchSnapshot();
  });

  test('корректно отображает кнопку без текста', () => {
    const {container} = render(<Button/>);
    expect(container).toMatchSnapshot();
  });

  test('корректно отображает отключенную кнопку', () => {
    const {container} = render(<Button disabled/>);
    expect(container).toMatchSnapshot();
  });

  test('корректно отображает кнопку с индикацией загрузки', () => {
    const {container} = render(<Button isLoader/>);
    expect(container).toMatchSnapshot();
  });

  test('корректность вызова колбека при клике на кнопку', () => {
    const handleClick = jest.fn();
    const {getByText} = render(<Button text="Click me!" onClick={handleClick}/>);
    getByText('Click me!').click();
    expect(handleClick).toHaveBeenCalled();
  });
});
