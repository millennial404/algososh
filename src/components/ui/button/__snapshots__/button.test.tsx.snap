// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component корректно отображает кнопку без текста 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</div>
`;

exports[`Button component корректно отображает кнопку с индикацией загрузки 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</div>
`;

exports[`Button component корректно отображает кнопку с текстом 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Click me!
    </p>
  </button>
</div>
`;

exports[`Button component корректно отображает отключенную кнопку 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</div>
`;
